### ISSUE TITLE
Add another dataloader project and example to docs page

### ISSUE THREAD

**Issue by frederikhors** [99designs/gqlgen#1714]
Given the (legitimate) absence of @vektah from his creations ([gqlgen](https://github.com/99designs/gqlgen) and [dataloaden](https://github.com/vektah/dataloaden)) and the difficulties we are having in keeping standing at least `gqlgen` I tried to use [another dataloader](https://github.com/graph-gophers/dataloader) (which did not have the `dataloaden` bugs) and I must say that **it works great**.

**It works without generation and without reflection** and has many options!

In addition I found an extremely nice person, @zenyui, who helped me to use `graph-gophers/dataloader` with `gqlgen`; here is a concrete and working example of integration: https://github.com/zenyui/gqlgen-dataloader.

What do you think to add in the [dataloader page in the docs](https://gqlgen.com/reference/dataloaders/) also this dataloader with the @zenyui's example?

Plurality has always been growth.

Thanks for your work.

**Comment by zenyui** [comment]
Happy to help! I found the `graph-gophers/dataloader` easy to implement and, feature-wise, a drop-in replacement for dataloaden.

**Comment by MiguelCastillo** [comment]
FWIW - we use github.com/graph-gophers/dataloader at Khan Academy with success.

**Comment by StevenACoffman** [comment]
Thanks! Please make a PR with the documentation change you mentioned.

**Comment by zenyui** [comment]
@StevenACoffman nice, i'll collab with @frederikhors on it.

**Comment by frederikhors** [comment]
@zenyui have you an idea on a PR?

**Comment by zenyui** [comment]
@frederikhors yeah, my place is to open a PR to replace the dataloader docs with a simple middleware definition as code snippets and a link to the sample app. Happy to collaborate on it (messaged you on discord)

**Comment by frederikhors** [comment]
@zenyui. Ok. My opinion is not to remove the old dataloader anyway. I wish it was clear in the docs that there are at least two valid approaches.

**Comment by zenyui** [comment]
@frederikhors good idea, i'll do that

**Comment by StevenACoffman** [comment]
@zenyui Hey, how's this going?

**Comment by zenyui** [comment]
hey! got pulled away from this. think i can wrap it up this weekend

**Comment by zenyui** [comment]
@StevenACoffman @frederikhors alrighty I opened a PR that updates the docs to use graph-gophers/dataloader.

https://github.com/99designs/gqlgen/pull/1871

I know we discussed having both options in parallel... I originally did that and it made the docs too hard to understand. IMO the graph-gophers/dataloader library should be the default/recommended library because:
- it has been more recently maintained
- it does not require codegen, and the only benefit of code gen is the final type casting

happy to get your thoughts here, on the PR comments, or in discord. LMK your thoughts!
